codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoJmespathExpressionGenerator.java

```diff
@@ -420,10 +458,20 @@ private GoWriter.Writable compareVariables(String ident, Variable left, Variable
                         "cmp", cmp,
                         "lhs", isLPtr ? "*" + left.ident : left.ident,
                         "rhs", isRPtr ? "*" + right.ident : right.ident,
-                        "cast", cast
+                        "cast", cast,
+                        "nilCoerceLeft", nilCoerceLeft,
+                        "nilCoerceRight", nilCoerceRight
+                ),
+                Map.of(
+                        "elseCheckPtrs", elseCheckPtrs
                 ));
     }
 
+    private static boolean isOrderComparator(ComparatorType cmp) {
```

Comment:
rant (?), non-blocking: I wouldn't know how to name this function so I would get "comparisons excluding direct equality check" without reading inside it. TIL they are technically called "inequalities"[link](https://en.wikipedia.org/wiki/Inequality_(mathematics)) but I'd definitely be thinking `!=` with that name. 

(order comparator is fine BTW)